appgate/attrs.py

```diff
@@ -79,7 +89,57 @@ def dump_datetime(v: datetime.datetime) -> str:
     return v.isoformat(timespec="milliseconds").replace("+00:00", "Z")
 
 
-def get_dumper(platform_type: PlatformType):
+def k8s_name(name: str) -> str:
+    # This is ugly but we need to go from a bigger set of strings
+    # into a smaller one :(
+    return re.sub("[^a-z0-9-.]+", "-", name.strip().lower())
+
+
+def k8s_dumper(
+    dumper: Dumper,
+    entity: Entity_T,
+    api_spec: APISpec,
+    strict: bool = True,
+    resolution_conflicts: Dict[str, List[MissingFieldDependencies]] | None = None,
+) -> Dict[str, Any]:
+    entity_kind = entity.__class__.__qualname__
+    annotations = {}
+    # Check if entity has some resolution conflicts
+    fs = ";".join(
+        f.field_path for f in (resolution_conflicts or {}).get(entity.name, [])
+    )
```

Comment:
Repeating our offline discussion. 

There is a 63 character limit for annotation. I wonder if there are any entities with enough id fields that could hit this limit.

Reply:
Probably we can have just one annotation per field, this is mostly to know what fields were not able to resolve the ids into names.

Reply:
I will add a TODO there